src/torchmetrics/text/infolm.py

```diff
@@ -0,0 +1,198 @@
+        temperature:
+            A temperature for calibrating language modelling. For more information, please reference `InfoLM`_ paper.
+        information_measure:
+            A name of information measure to be used. Please use one of: ['kl_divergence', 'alpha_divergence',
+            'beta_divergence', 'ab_divergence', 'renyi_divergence', 'l1_distance', 'l2_distance', 'l_infinity_distance',
+            'fisher_rao_distance']
+        idf:
+            An indication of whether normalization using inverse document frequencies should be used.
+        alpha:
+            Alpha parameter of the divergence used for alpha, AB and Rényi divergence measures.
+        beta:
+            Beta parameter of the divergence used for beta and AB divergence measures.
+        device:
+            A device to be used for calculation.
+        max_length:
+            A maximum length of input sequences. Sequences longer than `max_length` are to be trimmed.
+        batch_size:
+            A batch size used for model processing.
+        num_threads:
+            A number of threads to use for a dataloader.
+        verbose:
+            An indication of whether a progress bar to be displayed during the embeddings calculation.
+        return_sentence_level_score:
+            An indication whether a sentence-level InfoLM score to be returned.
+
+    Example:
+        >>> from torchmetrics.text.infolm import InfoLM
+        >>> preds = ['he read the book because he was interested in world history']
+        >>> target = ['he was interested in world history because he read the book']
+        >>> infolm = InfoLM('google/bert_uncased_L-2_H-128_A-2', idf=False)
+        >>> infolm(preds, target)
+        tensor(-0.1784)
+
+    References:
+        [1] InfoLM: A New Metric to Evaluate Summarization & Data2Text Generation by Pierre Colombo, Chloé Clavel and
+        Pablo Piantanida `InfoLM`_
+    """
+
+    is_differentiable = False
+    higher_is_better = True
+    preds_input_ids: List[Tensor]
+    preds_attention_mask: List[Tensor]
+    target_input_ids: List[Tensor]
+    target_attention_mask: List[Tensor]
+
+    def __init__(
+        self,
+        model_name_or_path: Union[str, os.PathLike] = "bert-base-uncased",
+        temperature: float = 0.25,
+        information_measure: _ALLOWED_INFORMATION_MEASURE_LITERAL = "kl_divergence",
+        idf: bool = True,
+        alpha: Optional[float] = None,
+        beta: Optional[float] = None,
+        device: Optional[Union[str, torch.device]] = None,
+        max_length: Optional[int] = None,
+        batch_size: int = 64,
+        num_threads: int = 4,
+        verbose: bool = True,
+        return_sentence_level_score: bool = False,
+        compute_on_step: Optional[bool] = None,
+        **kwargs: Dict[str, Any],
+    ):
+        super().__init__(compute_on_step=compute_on_step, **kwargs)
+        self.model_name_or_path = model_name_or_path
```

Comment:
Question: should we have such a check?
Or does it only depends on transformers in specific cases?
```suggestion
        if not _TRANSFORMERS_AUTO_AVAILABLE or not _TRANSFORMERS_AVAILABLE:
            raise ValueError("Install stuff ...")
        self.model_name_or_path = model_name_or_path
```